### PR TITLE
Add some documentation

### PR DESCRIPTION
**Pull request by iTrooz**
This PR adds some documentation. It's actually pretty random, I followed the function calls I was curious about and commented whenever I wasn't sure/I thought it needed clarification

You might want to make sure to squash them, because the commits are kind of a mess, I didn't went through the effort of interactive rebase

### CODE REVIEW

**Comment by iTrooz** [comment]
Draft because I'm not sure I understand this code: https://github.com/WerWolv/ImHex/blob/ddcba81ed14689584481d1faddd149832f0e52bb/plugins/builtin/source/content/welcome_screen.cpp#L508

File: lib/libimhex/include/hex/api/imhex_api.hpp [hunk 1]
```diff
@@ -225,7 +225,11 @@ namespace hex {
 
         }
 
-        /* Functions to interact with the loaded data provider */
+        /**
+         * Helper methods about the providers
+         * NOTE: the "current provider" or "currently selected provider" refers to the currently selected provider in the UI;
+         * the provider is user is actually editing.
```
Comment (WerWolv) [inline lib/libimhex/include/hex/api/imhex_api.hpp:231]: Typo I think?

File: lib/libimhex/include/hex/api/imhex_api.hpp [hunk 2]
```diff
@@ -261,26 +265,26 @@ namespace hex {
...
              */
             bool isDirty();
 
 
             /**
-             * @brief Adds a newly created provider to the list of providers
+             * @brief Adds a newly created provider to the list of providers, and mark it as the selected one.
              * @param provider The provider to add
-             * @param skipLoadInterface Whether to skip loading the provider's loading interface
+             * @param skipLoadInterface Whether to skip the provider's loading interface (see property documentation)
```
Comment (WerWolv) [inline lib/libimhex/include/hex/api/imhex_api.hpp:287]: Did you mean "Provider documentation" here?

File: lib/libimhex/include/hex/api/imhex_api.hpp [hunk 3]
```diff
@@ -302,9 +306,9 @@ namespace hex {
             void remove(prv::Provider *provider, bool noQuestions = false);
 
             /**
-             * @brief Creates a new provider using its unlocalized name
+             * @brief Creates a new provider using its unlocalized name and add it to the list of providers
              * @param unlocalizedName The unlocalized name of the provider to create
-             * @param skipLoadInterface Whether to skip loading the provider's loading interface
+             * @param skipLoadInterface Whether to skip the provider's loading interface (see property documentation)
```
Comment (WerWolv) [inline lib/libimhex/include/hex/api/imhex_api.hpp:311]: Same here

File: lib/libimhex/include/hex/providers/provider.hpp [hunk 4]
```diff
@@ -96,6 +96,13 @@ namespace hex::prv {
         [[nodiscard]] virtual std::vector<std::pair<std::string, std::string>> getDataDescription() const = 0;
         [[nodiscard]] virtual std::variant<std::string, i128> queryInformation(const std::string &category, const std::string &argument);
 
+        /**
+         * @brief opens this provider
+         * the return value of this function allows to ensure the provider is available,
+         * so calling Provider::isAvailable() just after a call to open() that returned true is dedundant.
+         * NOTE: This is not related to the EventProviderOpened event
```
Comment (WerWolv) [inline lib/libimhex/include/hex/providers/provider.hpp:103]: These `NOTE:` can be replaced with the proper doxygen command `@note`

File: lib/libimhex/include/hex/providers/provider.hpp [hunk 5]
```diff
@@ -145,7 +152,19 @@ namespace hex::prv {
 
         u32 m_id;
 
+        /**
+         * @brief true if there is any data that needs to be saved
```
Comment (WerWolv) [inline lib/libimhex/include/hex/providers/provider.hpp:156]: These members are internal so they don't _really_ need any documentation. Nobody is able to use them anyways

File: plugins/builtin/source/content/welcome_screen.cpp [hunk 6]
```diff
@@ -409,7 +409,10 @@ namespace hex::plugin::builtin {
         }
         ImGui::End();
     }
-
+    /**
+     * @brief draw some default background if there are no views avaialble in the current layout
+     * TODO add a button to load the default layout ?
```
Comment (WerWolv) [inline plugins/builtin/source/content/welcome_screen.cpp:414]: Todos should probably not be in here

File: plugins/builtin/source/content/welcome_screen.cpp [hunk 7]
```diff
@@ -431,10 +434,16 @@ namespace hex::plugin::builtin {
...
+    /**
+     * @brief registers the event handlers related to the welcome screen
+    * should only be called once, at startup
+     */
     void createWelcomeScreen() {
         updateRecentProviders();
 
         (void)EventManager::subscribe<EventFrameBegin>(drawWelcomeScreen);
+
+        // sets a background when they are no views
```
Comment (WerWolv) [inline plugins/builtin/source/content/welcome_screen.cpp:446]: Try to use capital letters at the start of comments

**Review by WerWolv — CHANGES_REQUESTED** [review]

File: lib/libimhex/include/hex/api/imhex_api.hpp (hunk 2 above)
Comment (iTrooz) [reply lib/libimhex/include/hex/api/imhex_api.hpp:287]: I was referring to the `m_skipLoadInterface` property: https://github.com/WerWolv/ImHex/blob/4912420d7eff8dc38a974158893c87355cb74726/lib/libimhex/include/hex/providers/provider.hpp#L160

What would you write ?

File: plugins/builtin/source/content/welcome_screen.cpp (hunk 6 above)
Comment (iTrooz) [reply plugins/builtin/source/content/welcome_screen.cpp:414]: Oh yeah, issues are probably better

File: plugins/builtin/source/content/welcome_screen.cpp (hunk 7 above)
Comment (iTrooz) [reply plugins/builtin/source/content/welcome_screen.cpp]: Doing that now !

**Review by WerWolv — APPROVED** [review]

**Pull request merged**